examples/kubernetes_orchestration/README.md

```diff
@@ -0,0 +1,194 @@
+# :dango: Pipeline Orchestration on Kubernetes
+
+This example will demonstrate how to orchestrate pipelines using the ZenML
+Kubernetes-native orchestrator, which is a lightweight alternative to other
+distributed orchestrators like Airflow or Kubeflow.
+
+Overall, the Kubernetes orchestrator is quite similar to the Kubeflow
+orchestrator in that it runs each step of the pipeline in a separate
+Kubernetes pod. However, the orchestration of the different pods is not done
+by Kubeflow, but by a separate master pod that orchestrates the step execution
+via topological sort.
+
+In this example, we will build a simple pipeline consisting of four steps and
+orchestrate it in a Kubernetes cluster running in the cloud.
+
+## :heavy_check_mark: Requirements
+
+In this example, we will use an AWS cloud stack, consisting of an EKS cluster,
+an ECR container registry, and a S3 bucket for artifact storage.
+
+If you want to follow this example line by line, you need to spin up each of
+the corresponding AWS services first.
+Alternatively, you can can also use any other cloud provider and adjust all
+`zenml ... register` commands below accordingly.
+
+Regardless of your cloud provider choice, you will also need to have kubectl
+and Docker installed locally.
+
+### Setup and Register Kubernetes Orchestrator
+After spinning up your Kubernetes cluster in the cloud, you will first need
+to connect it to your local kubectl as a new kubernetes context.
+
+How to do this depends on your cloud provider. For AWS EKS, you can do the
+following:
+
+```bash
+aws eks --region <AWS_REGION> update-kubeconfig
+    --name <AWS_EKS_CLUSTER>
+    --alias <KUBE_CONTEXT>
+```
+
+**Note:** It does not matter what you use as `KUBE_CONTEXT` here, as long as it
+is a unique name.
+
+Next, register your Kubernetes orchestrator with the respective kubernetes 
+context:
+
+```bash
+zenml orchestrator register k8s_orchestrator
+    --flavor=kubernetes
+    --kubernetes_context=<KUBE_CONTEXT>
+    --synchronous=True
+```
+
+### Setup and Register Metadata Store
+
+If you want to store your metadata locally within the kubernetes cluster, you
+can use the `KubernetesMetadataStore`, which is a subclass of the
+`MySQLMetadataStore` that will automatically spin up a MySQL deployment in
+your cluster if you call `zenml stack up`.
+
+We can register this metadata store as follows:
+
+```bash
+zenml metadata-store register kubernetes_store 
+    --flavor=kubernetes
+    --host=mysql
+    --port=3306
+    --database=mysql
+    --username=root
```

Comment:
I could be wrong, but I didn't see you using these configuration values when setting up the MySQL deployment. If I passed here a different host, port, database or username value, won't that break the metadata store ?Does that mean that these values would better be modeled as constants instead of configurable attributes ?

I know you modeled this metadata store based on the mysql one, and there these values could in fact be customized, but it doesn't look like that's the case here. I don't know what's a good quick fix for this, maybe just don't mention these values in the docs and hope the users don't try to customize them :)